### ISSUE TITLE
Not segmenting if image size is less than 600x600

### ISSUE THREAD

**Issue by srika91** [tmbdev/ocropy#88]
The first part of binarization is working correctly for my image.But in the segmentation step if both  or one of the height or width is less than 600x600, ocropus segmentation is not segmenting the binarized image. Is it necessary to have image > = 600x600, because upsampling the image makes the ocr part miserable. And also why is this 600x600 limitation? I am giving a cropped input to ocropus, so resizing is a problem.

**Comment by kendemu** [comment]
I am upscaling 480p video frame to 300dpi and have no problem with the ocr. If you want a high quality upscaling, use [waifu2x](https://github.com/nagadomi/waifu2x).

**Comment by zuphilip** [comment]
In the code you can deactivate this test by changing the [line 385 in toplevel.py](https://github.com/tmbdev/ocropy/blob/master/ocrolib/toplevel.py#L385) to
```py
    return a.ndim==2# and a.shape[0]>=600 and a.shape[1]>=600
```
It might be that this assumption was made also later in the code and therefore some tools or function are not working as expected. If someone has deeper insides or maybe experiences with the image size assumption, that would be helpful.

**Comment by amitdo** [comment]
From `ocropus-gpageseg`:

```
def check_page(image):
    if len(image.shape)==3: return "input image is color image %s"%(image.shape,)
    if mean(image)<median(image): return "image may be inverted"
    h,w = image.shape
    if h<600: return "image not tall enough for a page image %s"%(image.shape,)
    if h>10000: return "image too tall for a page image %s"%(image.shape,)
    if w<600: return "image too narrow for a page image %s"%(image.shape,)
    if w>10000: return "line too wide for a page image %s"%(image.shape,)
    ...
```

**Comment by tmbdev** [comment]
The text recognizer requires 300 DPI, and the page segmenter requires full pages from articles or books. The check is there because the kind of image you give it doesn't conform to these requirements and you need to do something to make it work.

**Comment by tmbdev** [comment]
You could try to upsample your image and then put it into the center of a large blank page so that both requirements are met. Unfortunately, there are multiple parameters involved (resolution and page size), so it requires since intelligence to figure out want to do.

Future recognizers will try to deal with this more intelligently.

**Comment by amitdo** [comment]
Tom,
Does your new project will be written in Python or C++ or maybe in Lua ? Are you going to use a framework like TensorFlow / Theano / Torch ?

**Comment by tmbdev** [comment]
I'm planning on using one of the existing frameworks. It's also going to be command line tools that should just drop in and replace corresponding tools in ocropy.

(I already have a couple of experimental TensorFlow implementations, but TensorFlow has a lot of overhead.)

**Comment by zuphilip** [comment]
Well, it is possible to deactivate the first checks in `gpageseg` @amitodo referenced, but this will still do the second tests as I showed. I tried the examples in the `test` directory and they work for me, also the croped images and the single text lines are working for `gpageseg`, e.g.

```
root@30d28daa821a:/ocropy# ./ocropus-gpageseg -n tests/text-near-edge.bin.png
INFO:
INFO:  ########## ./ocropus-gpageseg -n tests/text-near-edge.bin.png
INFO:
INFO:  tests/text-near-edge.bin.png
INFO:  scale 13.8564064606
INFO:  computing segmentation
INFO:  computing column separators
INFO:  computing lines
INFO:  propagating labels
INFO:  spreading labels
INFO:  number of lines4
INFO:  finding reading order
INFO:  writing lines
INFO:       3 tests/text-near-edge.bin.png 13.9  4
root@30d28daa821a:/ocropy# ./ocropus-rpred tests/text-near-edge/*.bin.png
INFO:
INFO:  ########## ./ocropus-rpred tests/text-near-edge/010001.bin.png tests/te
INFO:
INFO:  #inputs4
# loading object /usr/local/share/ocropus/en-default.pyrnn.gz
INFO:  tests/text-near-edge/010001.bin.png:O1inton Street, aouth from LIYingston
 Street.
INFO:  tests/text-near-edge/010002.bin.png:P. L. Sperr.
INFO:  tests/text-near-edge/010003.bin.png:NO REPODUCTIONS.
INFO:  tests/text-near-edge/010004.bin.png:Auguat S, 1934.
root@30d28daa821a:/ocropy#
```
Is there any need for the second test?

**Comment by amitdo** [comment]
@zuphilip, who is amit**o**do? :)

**Comment by zuphilip** [comment]
(Sorry, typo, corrected now.)

**Comment by zuphilip** [comment]
Fixed in #220.
